futures-executor/src/local_pool.rs

```diff
@@ -334,6 +337,78 @@ impl<S: Stream + Unpin> Iterator for BlockingStream<S> {
     }
 }
 
+/// Turn a sink into a blocking sink.
+pub fn block_on_sink<S: Sink<Item> + Unpin, Item>(sink: S) -> BlockingSink<S, Item> {
+    BlockingSink { sink, _phantom: PhantomData }
+}
+
+/// A sink which blocks on values from a sink until they become available.
+#[derive(Debug)]
+pub struct BlockingSink<S: Sink<Item> + Unpin, Item> {
+    sink: S,
+    _phantom: PhantomData<fn(Item)>,
+}
+
+impl<S: Sink<Item> + Unpin, Item> Unpin for BlockingSink<S, Item> {}
+
+impl<S: Sink<Item> + Unpin, Item> Deref for BlockingSink<S, Item> {
+    type Target = S;
+    fn deref(&self) -> &Self::Target {
+        &self.sink
+    }
+}
+
+impl<S: Sink<Item> + Unpin, Item> DerefMut for BlockingSink<S, Item> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.sink
+    }
+}
+
+impl<S: Sink<Item> + Unpin, Item> BlockingSink<S, Item> {
+    /// Convert this `BlockingSink` into the inner `Sink` type.
+    pub fn into_inner(self) -> S {
+        self.sink
+    }
+
+    /// Sends a value to this sink, blocking the current thread until it's able
+    /// to do so.
+    ///
+    /// This function will take the `value` provided and call the underlying
+    /// sink's `start_send` function until it's ready to accept the value. If
+    /// the function returns `Pending` then the current thread is blocked
+    /// until it is otherwise ready to accept the value.
+    ///
+    /// # Return value
+    ///
+    /// If `Ok(())` is returned then the `value` provided was successfully sent
+    /// along the sink, and if `Err(e)` is returned then an error occurred
+    /// which prevented the value from being sent.
+    pub fn send(&mut self, item: Item) -> Result<(), S::Error> {
+        LocalPool::new().run_until(self.sink.send(item))
```

Comment:
Why not just `block_on`? That should be cheaper.

Reply:
But more of a general thing: I'm not sure if those wrappers are pullling their weight. Just using `block_on(sink.send(item))` is not much more boilerplate, works for all `Future`s, and doesn't require the original async type to get moved.

Reply:
> Why not just `block_on`? That should be cheaper.

I just used the same thing as block_on_stream...
https://github.com/rust-lang-nursery/futures-rs/blob/25d638d720bfa128ff4f934d1742bfedfed20c89/futures-executor/src/local_pool.rs#L334